landlab/components/carbonate/carbonate_producer.py

```diff
@@ -0,0 +1,219 @@
+#! /usr/bin/env python
+from contextlib import contextmanager
+
+import numpy as np
+from landlab import Component
+
+
+@contextmanager
+def set_numpy_err(*args, **kwds):
+    settings = np.seterr(*args, **kwds)
+    try:
+        yield settings
+    finally:
+        np.seterr(**settings)
+
+
+def smooth_heaviside(x, width=0.5, out=None):

```

Comment:
Add docstring for `smooth_heaviside()`? Also, if this fn is likely to be useful for other components, consider moving to utils.